app/services/katello/pulp3/migration.rb

```diff
@@ -16,9 +16,21 @@ class Migration
         Katello::Erratum
       ].freeze
 
-      def initialize(smart_proxy, repository_types = REPOSITORY_TYPES)
+      def initialize(smart_proxy, repository_types = nil)
+        if repository_types.nil?
+          @repository_types = smart_proxy.supported_pulp_types[:pulp3][:overriden_to_pulp2]
+        else
+          @repository_types = repository_types
+        end
+
+        if @repository_types.empty?
+          fail ::Katello::Errors::Pulp3MigrationError, _("Nothing to migrate.")
+        elsif (@repository_types - REPOSITORY_TYPES).any?
+          fail ::Katello::Errors::Pulp3MigrationError, _("Repository types %s cannot be migrated.") %
+            (@repository_types - REPOSITORY_TYPES).join(', ')
+        end
```

Comment:
Currently only the REPOSITORY_TYPES are supported for migration, wouldn't it make sense to do something like this:
@repository_types = REPOSITORY_TYPES

Shouldn't we throw an error if 

REPOSITORY_TYPES -  smart_proxy.supported_pulp_types[:pulp3][:overriden_to_pulp2]

returns anything?  Remember that overriden to pulp2 is essentially telling katello what hasn't been migrated yet, so i think what you have is erroring if something hasn't been migrated yet, but wasn't requested for migration, which i don't think is a situation we need to be checking.

Reply:
What do you mean by "requested for migration"? I thought the types in `smart_proxy.supported_pulp_types[:pulp3][:overriden_to_pulp2]` were the types being requested for migration.

If only 'file' is in `smart_proxy.supported_pulp_types[:pulp3][:overriden_to_pulp2]`, I take that as meaning we should only migrate file.  Wouldn't `REPOSITORY_TYPES - smart_proxy.supported_pulp_types[:pulp3][:overriden_to_pulp2]` in that case return 'docker' and fail?

Reply:
Do you think that the migration should silently pass if there's nothing to migrate? Then the only error case would be if something is requested to be migrated that can't be.  That would just mean I take out the `@repository_types.empty?` check.